operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsRetriever.java

```diff
@@ -319,46 +401,182 @@ public WlsDomainConfig readConfig() {
+   * succeeded or the specified timeout period has reached.
+   *
+   * @param url The URL of the HTTP post request to be invoked
+   * @param payload The payload of the HTTP Post request to be invoked
+   * @param timeoutMillis Timeout in milliseconds
+   * @return The Json string returned from the HTTP POST request
+   * @throws Exception Any exception thrown while trying to invoke the HTTP POST request
+   */
+  private String executePostUrlWithRetry(final String url, final String payload, final long timeoutMillis)
+          throws Exception {
+    LOGGER.entering();
+
+    long stopTime = System.currentTimeMillis() + timeoutMillis;
+    Exception exception = null;
+    String jsonResult = null;
     long timeRemaining = stopTime - System.currentTimeMillis();
     // keep trying and ignore exceptions until timeout.
-    while (timeRemaining > 0 && result == null) {
+    while (timeRemaining > 0 && jsonResult == null) {
       LOGGER.info(MessageKeys.WLS_CONFIGURATION_READ_TRYING, timeRemaining);
       exception = null;
       try {
         String serviceURL = connectAndGetServiceURL();
-        String jsonResult =
-          httpClient.executePostUrlOnServiceClusterIP(WlsDomainConfig.getRetrieveServersSearchUrl(), serviceURL, WlsDomainConfig.getRetrieveServersSearchPayload()).getResponse();
-        if (jsonResult != null) {
-          result = WlsDomainConfig.create().load(jsonResult);
-        }
+        jsonResult = httpClient.executePostUrlOnServiceClusterIP(url, serviceURL, payload).getResponse();
       } catch (Exception e) {
         exception = e;
         LOGGER.info(MessageKeys.WLS_CONFIGURATION_READ_RETRY, e, READ_CONFIG_RETRY_MILLIS);
       }
-      try {
-        // sleep before retrying
-        Thread.sleep(READ_CONFIG_RETRY_MILLIS);
-      } catch (InterruptedException ex) {
-        // ignore
+      if (jsonResult == null) {
+        try {
+          // sleep before retrying
+          Thread.sleep(READ_CONFIG_RETRY_MILLIS);
+        } catch (InterruptedException ex) {
+          // ignore
+        }
       }
       timeRemaining = stopTime - System.currentTimeMillis();
     }
-    if (result == null && exception != null) {
+    if (jsonResult == null && exception != null) {
       LOGGER.throwing(exception);
       throw exception;
     }
+    LOGGER.exiting(jsonResult);
+    return jsonResult;
+  }
+
+  /**
+   * Update the dynamic cluster size of the WLS cluster configuration.
+   *
+   * @param wlsClusterConfig The WlsClusterConfig object of the WLS cluster whose cluster size needs to be updated
+   * @param machineNamePrefix Prefix of names of new machines to be created
+   * @param targetClusterSize The target dynamic cluster size
+   * @return true if the request to update the cluster size is successful, false if it was not successful within the
+   *         time period, or the cluster is not a dynamic cluster
+   */
+  public boolean updateDynamicClusterSize(final WlsClusterConfig wlsClusterConfig,
```

Comment:
Who calls this?  We should discuss...  Other places use a pattern where this is wrapped in a Step which suspends until the asynchronous activity finishes.  This allows for timeout and retry, etc.

Reply:
There is no usages. It was added some time ago when we thought the RestBackEndImpl class may need to adjust the dynamic cluster size. I will remove this method.